tempesta_fw/ss_skb.c

```diff
@@ -297,25 +367,20 @@ __split_linear_data(struct sk_buff *skb, char *pspt, int len, TfwStr *it)
 	BUG_ON(-len > tail_len);
 
 	/*
-	 * Quick and unlikely path: just advance the skb tail pointer.
-	 * Note that this only works when we make room. When we remove,
-	 * pspt points at the start of the data chunk to remove. In that
-	 * case, tail_len can never be zero.
-	 */
-	if (unlikely(!tail_len && len <= ss_skb_tailroom(skb))) {
-		BUG_ON(len < 0);
-		it->ptr = ss_skb_put(skb, len);
-		return 0;
-	}
```

Comment:
Hm, it seems you moved the fast path to `__split_pgfrag_add()` and this doesn't look correct. See comment in `__split_pgfrag_add()`.